uaclient/data_types.py

```diff
@@ -50,6 +61,22 @@ def from_value(val: Any) -> Any:
         return val
 
 
+class EnumDataValue(DataValue):
+    """
+    To be used for parsing enum values
+    from_value raises an error if the value is not in the enum class values
+    and returns the value if found.
+    """
```

Comment:
Let's include something here about how to correctly create an EnumDataValue enum. Importantly, we should specify to inherit from both `EnumDataValue` and `Enum`

Something like:
```
To define a new EnumDataValue, create a new class that inherits from both `EnumDataValue` and `Enum`. E.g.
    class MyEnum(EnumDataValue, Enum):
        ONE = 1
        TWO = 2
```

Reply:
I think adding `Enum` to the class is a better approach. So that usage is only inheriting from EnumDataValue.